src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1082,6 +1083,7 @@ def dbapi(
         query_timeout: Optional[int] = 0,
         fetch_size: Optional[int] = 0,
         custom_schema: Optional[Union[str, StructType]] = None,
+        predicates: Optional[List[str]] = None,
```

Comment:
could you check what if predicates and columns are both passed in spark -- which one would takes priority or would spark error out?

Reply:
in pyspark, predicates are only allowed when columns is None, if both of them are not None, predicates will be ignored